handler/preprocess.sh

```diff
@@ -35,6 +36,48 @@ echo "finding dicom directories"
 ./find_dicomdir.py $root > $root/dcm2niix.list
 cat $root/dcm2niix.list
 
+# There can be overlap between the dicom directories and the pet directories
+# we want to use dcm2niix4pet for the pet directories, and dcm2niix for the dicom directories
+# but if and only if the conversion library PET2BIDS is installed
+PET2BIDS_INSTALLED=`which dcm2niix4pet > /dev/null && echo $?`
+
+if [ $PET2BIDS_INSTALLED -eq 0 ]; then
+
+    echo "PET2BIDS is installed, using dcm2niix4pet for PET directories found in root dir ${root}"
+    function rundcm2niix4pet {
+        #note.. this function runs inside $root (by --wd $root)
+
+        path=$1
+
+        echo "----------------------- $path ------------------------"
+        timeout 3600 dcm2niix4pet $path
+
+        #all good
+        echo $path >> pet2bids.done
+    } 
+
+    export -f rundcm2niix4pet
+
+    # now we do a little magic to run pet2bids if it's installed
+    ./find_petdir.py $root > $root/pet2bids.list
+    echo "Found PET directories:"
+    cat $root/pet2bids.list
+
+    # remove pet directories from dcm2niix list
+    echo "Removing PET directories from dcm2niix list"
+    grep -v -x -f <(sort pet2bids.list) <(sort dcm2niix.list) > dcm2niix.list
+    
+    # run pet2bids
+    true > $root/pet2bids.done
```

Comment:
`true` here isn't required

Reply:
is there a reason to do this here, given the process is not done as the file name indicates?

Reply:
btw I see you may have repeated the same structures from other parts of the code, to keep it in line with the rest- so myself or Dan will fix the other parts in posterior PRs.

Reply:
> btw I see you may have repeated the same structures from other parts of the code, to keep it in line with the rest- so myself or Dan will fix the other parts in posterior PRs.

Yup yup yup, just trying to keep convention and maintain the peace. If you would prefer to refactor handler/preprocess.sh first, I can just update from there when ready.